ledger/participant-integration-api/src/main/scala/platform/store/dao/MeteredLedgerDao.scala

```diff
@@ -151,7 +152,10 @@ private[platform] class MeteredLedgerDao(ledgerDao: LedgerDao, metrics: Metrics)
 
   override def currentHealth(): HealthStatus = ledgerDao.currentHealth()
 
+  override def transactionsWriter: TransactionsWriter = ledgerDao.transactionsWriter
```

Comment:
In order to not make this public, can we instead add two methods to `LedgerDao` proxying the preparation and execution methods in this and use those?

Reply:
Good point. Done: created `prepareTransactionInsert` method in `LedgerDao`; `storeTransaction` already takes care of execution